### PR TITLE
Driver: do not create a dSYM bundle for static archives

### PR DESCRIPTION
**Pull request by compnerd**
Creation of a dSYM bundle for a static library is not supported.  Do not
create a job to create a dSYM bundle for a static library.

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please test

**Review by artemcm — APPROVED** [review]
The change LGTM.

**Comment by artemcm** [comment]
```
SwiftDriver/Jobs/Planning.swift:478:5: error: switch must be exhaustive
swift-driver/Sources/SwiftDriver/Jobs/Planning.swift:478:5: note: add missing case: '.none'
```
I'm really confused by this error, this is the enum type:
https://github.com/apple/swift-driver/blob/main/Sources/SwiftDriver/Driver/LinkKind.swift#L14

**Comment by cltnschlosser** [comment]
> ```
> SwiftDriver/Jobs/Planning.swift:478:5: error: switch must be exhaustive
> swift-driver/Sources/SwiftDriver/Jobs/Planning.swift:478:5: note: add missing case: '.none'
> ```
> 
> I'm really confused by this error, this is the enum type:
> 
> https://github.com/apple/swift-driver/blob/main/Sources/SwiftDriver/Driver/LinkKind.swift#L14

`linkerOutputType` is optional, so `.none` is nil in this case.
some version of switch from the last couple years allows you to do this:
```swift
enum MyEnum {
  case A
  case B
}
let optionalEnum: MyEnum?
switch optionalEnum {
  case .A:
  case .B:
  case .none:
}
```

where before you needed
```swift
switch optionalEnum {
  case .some(.A):
  case .some(.B):
  case .none:
}
```

**Comment by artemcm** [comment]
> `linkerOutputType` is optional, so `.none` is nil in this case.
> 
Ah, didn't notice the optional. Thanks, @cltnschlosser!

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by gottesmm** [comment]
@swift-ci test

**Comment by gottesmm** [comment]
compnerd and I looked at the error and the correct answer is indeed 3. You can see this since in the old driver this is how we used to emit it in terms of jobs:

**Comment by gottesmm** [comment]
@swift-ci test

**Comment by gottesmm** [comment]
This is the swift-driver version of apple/swift#37526

**Comment by gottesmm** [comment]
@artemcm what do you think about the Linux failure here? It doesn't find lib tool.

**Comment by compnerd** [comment]
@swift-ci please test

File: Tests/SwiftDriverTests/SwiftDriverTests.swift [hunk 1]
```diff
@@ -3066,6 +3066,24 @@ final class SwiftDriverTests: XCTestCase {
       XCTAssertFalse(plannedJobs.contains { $0.kind == .generateDSYM })
     }
 
+    do {
+      var env = ProcessEnv.vars
+      // As per Unix conventions, /var/empty is expected to exist and be empty.
+      // This gives us a non-existent path that we can use for libtool which
+      // allows us to run this this on non-Darwin platforms.
+      env["SWIFT_DRIVER_LIBTOOL_EXEC"] = "/var/empty/libtool"
```
Comment (artemcm) [inline Tests/SwiftDriverTests/SwiftDriverTests.swift:3074]: Hmm, I expected this to work.

File: Tests/SwiftDriverTests/SwiftDriverTests.swift [hunk 2]
```diff
@@ -3066,6 +3066,24 @@ final class SwiftDriverTests: XCTestCase {
...
+
+      // No dSYM generation (-g -emit-library -static)
+      var driver = try Driver(args: [
+        "swiftc", "-target", "x86_64-apple-macosx10.15", "-g", "-emit-library",
+        "-static", "-o", "library.a", "library.swift"
+      ])
```
Comment (artemcm) [inline Tests/SwiftDriverTests/SwiftDriverTests.swift:3080]:
```suggestion
      ], env: env))
```

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by gottesmm** [comment]
I am going to cherry-pick this to 5.5.

**Comment by compnerd** [comment]
@gottesmm ugh, I forgot to do that, thank you so much for staying on top of that!

**Pull request merged**